VideoApp/VideoApp/ViewControllers/Lobby/LobbyViewController.swift

```diff
@@ -24,16 +24,17 @@ class LobbyViewController: UIViewController {
     @IBOutlet weak var audioToggleButton: UIButton!
     @IBOutlet weak var videoToggleButton: UIButton!
     @IBOutlet weak var flipCameraButton: UIButton!
-    private let localParticipantFactory = LocalParticipantFactory()
+    private let roomFactory = RoomFactory()
     private let deepLinkStore: DeepLinkStoreWriting = DeepLinkStore.shared
     private let notificationCenter = NotificationCenter.default
-    private var participant: LocalParticipant!
+    private var room: Room!
+    private var participant: LocalParticipant { room.localParticipant }
     private var shouldRenderVideo = true
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        resetParticipant()
+        resetRoom()
```

Comment:
So after you end a call and leave the current `Room`, a new `Room` object is created, even if you are not actually in a `Room`? I'm worried that this will cause developers confusion because it is sort of blurring the line of what an SDK room is vs the app implementation of a room is and what it means.

I don't disagree with the app's `Room` having the Local Participant app object, but I am not sure how comfortable I feel with us using the room concept at the lobby screen. I know one of @aaalaniz concerns revolves around the apps providing constructs that are similar enough to the SDK but add their own subtle differences that sort of muddy the SDK concepts.

Reply:
Yes a new `VideoApp.Room` is created even when we aren't in a room but `VideoApp.Room` does not create a `TwilioVideo.Room` until `connect` is called on `VideoApp.Room`. 

I think it might be better if I rename `TwilioVideo.Room` to something like `RoomManager`. That we we aren't implying as much that there is a `TwilioVideo.Room`. Do you think that helps?

Reply:
`VideoApp.Room` represents comprehensive room lifecycle. It does work before connecting like allow for previewing local participant and fetching access token. This makes things simple for UI and allows a lot of flexibility with UI design. 

The lobby and room screen are currently implemented with two separate view controllers. The UI is so similar that it may be better (UX and architecture) to implement with one view controller. This would be simple to do with `VideoApp.Room` because the UI can access local participant, remote participants (empty if not connected), and connection status (disconnected, connecting, connected) regardless if there is an instance of `TwilioVideo.Room` running. This would be difficult to do in UI if `VideoApp.Room` did not represent comprehensive room lifecycle (only local participant > fetch access token, connecting > connected > local + remote participants).

Reply:
> I think it might be better if I rename `TwilioVideo.Room` to something like `RoomManager`. That way we aren't implying as much that there is a `TwilioVideo.Room` running all the time. Do you think that helps?

I don't think it a good idea to rename any of the SDK constructs. The idea of the app being open source is so that other developers get a window of how to use the SDK in their own apps. We don't want to hide or obfuscate the core SDK constructs from somebody looking at the app for reference. I need to think about this a bit more.

Reply:
The reason it is currently called `TwilioVideo.Room`...

There are I think 7 types right now in the app like this that sort of adapt an SDK type into something that the UI can easily use. And I just went with consistency...modeled my names after SDK and rely on Swift namespaces to distinguish. I thought it may be difficult to come up with unique names that are intuitive and not silly for all these types. 

So yeah not sure I like `RoomManager` better. And maybe it doesn't help with your actual concern.

Reply:
Ok yeah sounds good. It is a little tricky.